lib/cpp/src/thrift/server/TThreadPoolServer.cpp

```diff
@@ -146,9 +73,12 @@ void TThreadPoolServer::serve() {
       shared_ptr<TProcessor> processor = getProcessor(inputProtocol, outputProtocol, client);
 
       // Add to threadmanager pool
-      shared_ptr<TThreadPoolServer::Task> task(
-          new TThreadPoolServer::Task(*this, processor, inputProtocol, outputProtocol, client));
-      threadManager_->add(task, timeout_, taskExpiration_);
+      shared_ptr<TConnectedClient> pClient(
```

Comment:
I think you are referring to shared_from_this.  Here are some of the boost docs talking about the difference:
http://www.boost.org/doc/libs/1_57_0/libs/smart_ptr/enable_shared_from_this.html
http://www.boost.org/doc/libs/1_57_0/libs/smart_ptr/make_shared.html